ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -402,5 +402,21 @@ private IEnumerable<BidEntity> GetSortBids(IEnumerable<BidEntity> bids, BidsSort
             }
         }
 
+        public async Task<Response> VoteForSupplier(VotingRequest votingRequest)
+        {
+            BidEntity bid = await _context.Bids.Where(b => b.Id == votingRequest.BidId).Include(b => b.CurrentProposals).FirstOrDefaultAsync().ConfigureAwait(false);
+            if (bid == null)
+            {
+                return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
+            }
+
+            IEnumerable<SupplierProposalEntity> propsalsToUpdate = bid.CurrentProposals.Where(propsal => votingRequest.VotedSuppliersIds.Contains(propsal.SupplierId));
```

Comment:
there will be at most one proposal here, so you can query with FirstOrDefaultAsync and get the Entity obj

Reply:
why only one ? I implemented it such that we are getting an array of supplier ids , so there could be more than one proposal ( proposal for supplier x and propals for supplier y , both are part of the ids array )